### PR TITLE
Use filepath.Join everywhere in cmd/build

### PR DESCRIPTION
**Pull request by kostola**
### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: cmd/build/binary.go [hunk 1]
```diff
@@ -42,8 +42,9 @@ func compile(ctx context.Context, cmd string, goos, goarch string) error {
 		"-X", fmt.Sprintf("'package-operator.run/internal/version.version=%s'", appVersion),
 	}
 
+	cmdPath := dotFilepathJoin("cmd", cmd)
 	err = shr.New(env).Run(
-		"go", "build", "--ldflags", strings.Join(ldflags, " "), "--trimpath", "--mod=readonly", "-o", dst, "./cmd/"+cmd,
+		"go", "build", "--ldflags", strings.Join(ldflags, " "), "--trimpath", "--mod=readonly", "-o", dst, cmdPath,
```
Comment (eqrx) [inline cmd/build/binary.go]: Not a file path, only use forward slashes

File: cmd/build/generate.go [hunk 2]
```diff
@@ -45,21 +45,23 @@ func (g Generate) All(ctx context.Context) error {
 }
 
 func (Generate) code() error {
+	const paths = "paths="
+
 	err := shr.New(sh.WithWorkDir("apis")).Run("controller-gen",
 		"crd:crdVersions=v1,generateEmbeddedObjectMeta=true",
-		"paths=./core/...",
-		"output:crd:artifacts:config=../config/crds",
+		paths+dotFilepathJoin("core", "..."),
```
Comment (eqrx) [inline cmd/build/generate.go]: Not a file path, only use forward slashes

File: cmd/build/generate.go [hunk 3]
```diff
@@ -45,21 +45,23 @@ func (g Generate) All(ctx context.Context) error {
...
+		"output:crd:artifacts:config="+filepath.Join("..", "config", "crds"),
 	)
 	if err != nil {
 		return fmt.Errorf("generating kubernetes manifests: %w", err)
 	}
 
-	err = sh.New(sh.WithWorkDir("apis")).Run("controller-gen", "object", "paths=./...")
+	err = sh.New(sh.WithWorkDir("apis")).Run("controller-gen", "object", paths+dotFilepathJoin("..."))
```
Comment (eqrx) [inline cmd/build/generate.go]: Not a file path, only use forward slashes

File: cmd/build/generate.go [hunk 4]
```diff
@@ -45,21 +45,23 @@ func (g Generate) All(ctx context.Context) error {
...
 	if err != nil {
 		return fmt.Errorf("generating deep copy methods: %w", err)
 	}
 
-	err = shr.Run("controller-gen", "object", "paths=./internal/...")
+	err = shr.Run("controller-gen", "object", paths+dotFilepathJoin("internal", "..."))
```
Comment (eqrx) [inline cmd/build/generate.go]: Not a file path, only use forward slashes

File: cmd/build/helpers.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package main
+
+import "path/filepath"
+
+func dotFilepathJoin(elem ...string) string {
```
Comment (eqrx) [inline cmd/build/helpers.go]: method only used for packages, not file paths, would say that there is no correct use for it

File: cmd/build/lint.go [hunk 6]
```diff
@@ -27,9 +27,17 @@ func (l Lint) goModTidyAll(ctx context.Context) error {
 }
 
 func (Lint) glciFix() error {
-	return shr.Run("golangci-lint", "run", "--fix", "--deadline=15m", "./...", "./apis/...", "./pkg/...")
+	return shr.Run("golangci-lint", "run", "--fix", "--deadline=15m",
+		dotFilepathJoin("..."),
```
Comment (eqrx) [inline cmd/build/lint.go]: Not a file path, only use forward slashes

File: cmd/build/lint.go [hunk 7]
```diff
@@ -27,9 +27,17 @@ func (l Lint) goModTidyAll(ctx context.Context) error {
...
+		dotFilepathJoin("apis", "..."),
```
Comment (eqrx) [inline cmd/build/lint.go]: Not a file path, only use forward slashes

File: cmd/build/lint.go [hunk 8]
```diff
@@ -27,9 +27,17 @@ func (l Lint) goModTidyAll(ctx context.Context) error {
...
+		dotFilepathJoin("pkg", "..."),
```
Comment (eqrx) [inline cmd/build/lint.go]: Not a file path, only use forward slashes

File: cmd/build/lint.go [hunk 9]
```diff
@@ -27,9 +27,17 @@ func (l Lint) goModTidyAll(ctx context.Context) error {
...
+	)
 }
 
 func (Lint) glciCheck() error {
-	return shr.Run("golangci-lint", "run", "--deadline=15m", "./...", "./apis/...", "./pkg/...")
+	return shr.Run("golangci-lint", "run", "--deadline=15m",
+		dotFilepathJoin("..."),
```
Comment (eqrx) [inline cmd/build/lint.go]: Not a file path, only use forward slashes

File: cmd/build/package.go [hunk 10]
```diff
@@ -27,9 +26,9 @@ func buildPackage(ctx context.Context, name, registry string) error {
 		}
 	}
 
-	path := fmt.Sprintf("./config/packages/%s/container.oci.tar", name)
+	path := dotFilepathJoin("config", "packages", name, "container.oci.tar")
```
Comment (eqrx) [inline cmd/build/package.go]: Does not need a dot in front of it

File: cmd/build/package.go [hunk 11]
```diff
@@ -27,9 +26,9 @@ func buildPackage(ctx context.Context, name, registry string) error {
...
 	err := cmd.NewBuild().BuildFromSource(ctx,
-		"./config/packages/"+name,
+		dotFilepathJoin("config", "packages", name),
```
Comment (eqrx) [inline cmd/build/package.go]: Does not need a dot in front of it

File: cmd/build/package.go [hunk 12]
```diff
@@ -41,7 +40,7 @@ func buildPackage(ctx context.Context, name, registry string) error {
 }
 
 func pushPackage(ctx context.Context, name, registry string) error {
-	imgPath, err := filepath.Abs("./config/packages/" + name)
+	imgPath, err := filepath.Abs(dotFilepathJoin("config", "packages", name))
```
Comment (eqrx) [inline cmd/build/package.go]: Does not need a dot in front of it

File: cmd/build/test.go [hunk 13]
```diff
@@ -143,15 +143,19 @@ func (t Test) Integration(ctx context.Context, self run.DependencyIDer, filter s
 		env["PKO_TEST_LATEST_BOOTSTRAP_JOB"] = url
 	}
 
+	coverPkg := dotFilepathJoin("...") + "," +
```
Comment (eqrx) [inline cmd/build/test.go]: Not a file path, only use forward slashes

File: cmd/build/test.go [hunk 14]
```diff
@@ -143,15 +143,19 @@ func (t Test) Integration(ctx context.Context, self run.DependencyIDer, filter s
...
+		dotFilepathJoin("apis", "...") + "," +
```
Comment (eqrx) [inline cmd/build/test.go]: Not a file path, only use forward slashes

File: cmd/build/test.go [hunk 15]
```diff
@@ -143,15 +143,19 @@ func (t Test) Integration(ctx context.Context, self run.DependencyIDer, filter s
...
+		dotFilepathJoin("pkg", "...")
```
Comment (eqrx) [inline cmd/build/test.go]: Not a file path, only use forward slashes

File: cmd/build/test.go [hunk 16]
```diff
@@ -143,15 +143,19 @@ func (t Test) Integration(ctx context.Context, self run.DependencyIDer, filter s
...
+
 	tArgs := []string{
 		"go", "test",
-		"-tags=integration", "-coverprofile=.cache/integration/cover.txt",
+		"-tags=integration", "-coverprofile=" + filepath.Join(".cache", "integration", "cover.txt"),
 		f, "-race", "-test.v", "-failfast", "-timeout=20m", "-count=1", "-json",
-		"-coverpkg=./...,./apis/...,./pkg/...", "./integration/...", "|", "gotestfmt", "--hide=empty-packages",
+		"-coverpkg=" + coverPkg, dotFilepathJoin("integration", "..."), "|", "gotestfmt", "--hide=empty-packages",
```
Comment (eqrx) [inline cmd/build/test.go]: Not a file path, only use forward slashes

File: cmd/build/test.go [hunk 17]
```diff
@@ -165,21 +169,23 @@ func (t Test) Integration(ctx context.Context, self run.DependencyIDer, filter s
...
 	}
 
-	gotestArgs := []string{"-coverprofile=.cache/unit/cover.txt", "-race", "-json"}
+	gotestArgs := []string{"-coverprofile=" + filepath.Join(".cache", "unit", "cover.txt"), "-race", "-json"}
 	if len(filter) > 0 {
 		gotestArgs = append(gotestArgs, "-run="+filter)
 	}
 
 	argStr := strings.Join(gotestArgs, " ")
+	tstPath := dotFilepathJoin("...")
```
Comment (eqrx) [inline cmd/build/test.go]: Not a file path, only use forward slashes

**Review by eqrx — CHANGES_REQUESTED** [review]

File: cmd/build/binary.go (hunk 1 above)
Comment (kostola) [reply cmd/build/binary.go]: Fixed

File: cmd/build/generate.go (hunk 2 above)
Comment (kostola) [reply cmd/build/generate.go]: Fixed

File: cmd/build/generate.go (hunk 3 above)
Comment (kostola) [reply cmd/build/generate.go]: Fixed

File: cmd/build/generate.go (hunk 4 above)
Comment (kostola) [reply cmd/build/generate.go]: Fixed

File: cmd/build/helpers.go (hunk 5 above)
Comment (kostola) [reply cmd/build/helpers.go]: Removed

File: cmd/build/lint.go (hunk 6 above)
Comment (kostola) [reply cmd/build/lint.go]: Fixed

File: cmd/build/lint.go (hunk 8 above)
Comment (kostola) [reply cmd/build/lint.go]: Fixed

File: cmd/build/lint.go (hunk 7 above)
Comment (kostola) [reply cmd/build/lint.go]: Fixed

File: cmd/build/lint.go (hunk 9 above)
Comment (kostola) [reply cmd/build/lint.go]: Fixed

File: cmd/build/package.go (hunk 10 above)
Comment (kostola) [reply cmd/build/package.go]: Fixed

File: cmd/build/package.go (hunk 11 above)
Comment (kostola) [reply cmd/build/package.go]: Fixed

File: cmd/build/package.go (hunk 12 above)
Comment (kostola) [reply cmd/build/package.go]: Fixed

File: cmd/build/test.go (hunk 13 above)
Comment (kostola) [reply cmd/build/test.go]: Fixed

File: cmd/build/test.go (hunk 14 above)
Comment (kostola) [reply cmd/build/test.go]: Fixed

File: cmd/build/test.go (hunk 15 above)
Comment (kostola) [reply cmd/build/test.go]: Fixed

File: cmd/build/test.go (hunk 16 above)
Comment (kostola) [reply cmd/build/test.go]: Fixed

File: cmd/build/test.go (hunk 17 above)
Comment (kostola) [reply cmd/build/test.go]: Fixed

File: cmd/build/test.go [hunk 18]
```diff
@@ -123,21 +123,22 @@ func (t Test) Integration(ctx context.Context, filter string) error {
...
 
 	argStr := strings.Join(gotestArgs, " ")
+	logPath := filepath.Join(".cache", "unit", "gotest.log")
 
 	return sh.New(
 		sh.WithEnvironment{"CGO_ENABLED": "1"},
 	).Bash(
 		"set -euo pipefail",
-		fmt.Sprintf("go test %s ./... 2>&1 | tee .cache/unit/gotest.log | gotestfmt --hide=empty-packages", argStr),
+		fmt.Sprintf("go test %s ./... 2>&1 | tee \"%s\" | gotestfmt --hide=empty-packages", argStr, logPath),
```
Comment (eqrx) [inline cmd/build/test.go]:
```suggestion
		fmt.Sprintf(`go test %s ./... 2>&1 | tee "%s" | gotestfmt --hide=empty-packages`, argStr, logPath),
```

File: cmd/build/lint.go [hunk 19]
```diff
@@ -31,5 +31,5 @@ func (Lint) glciFix() error {
 }
 
 func (Lint) glciCheck() error {
-	return shr.Run("golangci-lint", "run", "--deadline=15m", "./...", "./apis/...", "./pkg/...")
+	return shr.Run("golangci-lint", "run", "--fix", "--deadline=15m", "./...", "./apis/...", "./pkg/...")
```
Comment (eqrx) [inline cmd/build/lint.go]:
```suggestion
	return shr.Run("golangci-lint", "run", "--deadline=15m", "./...", "./apis/...", "./pkg/...")
```
Nope, the check fn is supposed to check only. If you want to get rid of it lets discuss it, but that change would just create a duplicate method :)

File: cmd/build/generate.go [hunk 20]
```diff
@@ -59,7 +59,7 @@ func (Generate) code() error {
 		return fmt.Errorf("generating deep copy methods: %w", err)
 	}
 
-	err = shr.Run("controller-gen", "object", "paths=./internal/...")
+	err = sh.New(sh.WithWorkDir("apis")).Run("controller-gen", "object", "paths=./...")
```
Comment (eqrx) [inline cmd/build/generate.go]: This changes functionality, is that intended?

**Review by eqrx — CHANGES_REQUESTED** [review]

File: cmd/build/test.go (hunk 18 above)
Comment (kostola) [reply cmd/build/test.go]: Done

File: cmd/build/lint.go (hunk 19 above)
Comment (kostola) [reply cmd/build/lint.go]: Sorry my bad. I restored the old behavior now.

File: cmd/build/generate.go (hunk 20 above)
Comment (kostola) [reply cmd/build/generate.go]: Sorry my bad. I restored the old behavior now.

**Review by eqrx — APPROVED** [review]

**Pull request merged**